### ISSUE TITLE
Crash (abort) on prompt piping output to `head`

### ISSUE THREAD

**Issue by scop** [jdx/mise#3332]
**Describe the bug**
Could be probably generalized, but I see mise crashing with SIGABRT when it's waiting for input to the "install community developed" prompt, while its stderr is redirected to stdout and piped to `head -n 1`.

It does not crash immediately, but after a second+ or so.

**To Reproduce**
```
$ set -o pipefail
$ time mise install 1password 2>&1 | head -n 1
mise WARN  ⚠️ 1password is a community-developed plugin – https://github.com/NeoHsu/asdf-1password-cli

real	0m1,382s
user	0m0,024s
sys	0m0,044s

$ echo $?
134
```

**Expected behavior**
No crash.

**`mise doctor` output**

```text
version: 2024.11.37 linux-x64 (7ac5ab3 2024-11-30)
activated: yes
shims_on_path: no

build_info:
  Target: x86_64-unknown-linux-gnu
  Features: DEFAULT, NATIVE_TLS, OPENSSL
  Built: Sat, 30 Nov 2024 01:03:00 +0000
  Rust Version: rustc 1.82.0 (f6e511eec 2024-10-15)
  Profile: release

shell:
  /bin/bash
  GNU bash, version 5.2.21(1)-release (x86_64-pc-linux-gnu)
  Copyright (C) 2022 Free Software Foundation, Inc.
  License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

  This is free software; you are free to change and redistribute it.
  There is NO WARRANTY, to the extent permitted by law.

dirs:
  cache: ~/.cache/mise
  config: ~/.config/mise
  data: ~/.local/share/mise
  shims: ~/.local/share/mise/shims
  state: ~/.local/state/mise

config_files:
  ~/.config/mise/config.toml

backends:
  aqua
  asdf
  cargo
  core
  gem
  go
  npm
  pipx
  spm
  ubi
  vfox

plugins:
  dprint  https://github.com/asdf-community/asdf-dprint#79889ec
  usage   https://github.com/jdx/mise-usage.git#fe3888a

toolset:
  aqua:astral-sh/uv@0.5.5
  aqua:cli/cli@2.63.0
  aqua:crate-ci/committed@1.1.1
  aqua:crate-ci/typos@1.28.1
  aqua:evilmartians/lefthook@1.8.4
  aqua:goreleaser/nfpm@2.41.1
  aqua:gotestyourself/gotestsum@1.12.0
  aqua:mvdan/sh@3.10.0
  asdf:dprint@0.47.5
  asdf:usage@1.3.3
  core:go@1.23.3
  ubi:aquaproj/aqua@2.38.1
  ubi:astral-sh/ruff@0.8.1
  ubi:golangci/golangci-lint@1.62.2
  ubi:goreleaser/goreleaser@2.4.8
  ubi:koalaman/shellcheck@0.10.0
  ubi:suzuki-shunsuke/cmdx@1.7.4

env_vars:
  MISE_USE_TOML=true
  MISE_SHELL=bash

settings:

1 warning found:

1. plugin python overrides a core plugin

No problems found

```

**Additional context**
Some possibly useful info grabbed from Ubuntu's crash viewer tool (screenshot, copying text out of the tool doesn't seem possible). Not sure how I'd go about getting the rust stack trace of this.

![Screenshot from 2024-12-03 08-05-33](https://github.com/user-attachments/assets/7bf34057-01c4-47a1-8001-65838a52bbc5)

**Comment by jdx** [comment]
mise has to crash in this scenario. It hasn't got the confirmation it needs and isn't able to show an error message either. You can pass `--yes` but I don't see what mise could do here that's better than the current state.

**Comment by scop** [comment]
In my opinion, it should continue waiting for the confirmation indefinitely.  

Or if it has to exit, exit gracefully with a nonzero exit status, instead of crashing.

Anyway also in my opinion, crashing is only appropriate when there is a programming error, or an assertion failure, severe enough to make continuing impossible, and crashing is a way to convey diagnostics about it so it can be fixed.

**Comment by jdx** [comment]
I certainly disagree. If mise just hung without doing anything that would be much worse.

**Comment by scop** [comment]
It's the way all other commands I can think of offhand that read stdin behave -- they just wait for stdin until something comes out of it, or it is closed.

**Comment by jdx** [comment]
this isn't really reading from stdin like most CLI tools, it's specifically prompting the user for input. If we're not able to ask the question, we shouldn't expect a response.

**Comment by scop** [comment]
That expectation and opinions on it aside, mine still is that crashing is not the appropriate action in this situation, and that's also the subject of this bug report (not what to do with stdin or stdout). Graceful exit with a nonzero exit status would be fine.

Crashes may fill disks and journals with crash dumps, provoke crash handling tools asking users to submit the data to vendors, etc etc. (Incidentally, that's how I found out about this crash, as the Ubuntu crash report tool started popping up when I happened to provoke the crash.)
